Scrape card data from dominionstrategy.com

mickvangelderen opened the pull request:
The data is not complete, which is why I've chosen to put the json file under
revision control. This allows us to apply manual fixes.

Next up is using the quote crate to actually generate the card data code.

mickvangelderen committed:
Scrape card data from dominionstrategy.com

The data is not complete, which is why I've chosen to put the json file under
revision control. This allows us to apply manual fixes.

Next up is using the quote crate to actually generate the card data code.

andebjor reviewed (commented):
Looks good, but I don't feel qualified to review this :/

oliverlee commented on dominion/Cargo.toml:
```diff
@@ -0,0 +1,14 @@
+[package]
+name = "dominion"
+version = "0.1.0"
+authors = ["Oliver Lee <oliverzlee@gmail.com>"]
```
Please add yourself (and Bjorn) while you're at it.

oliverlee commented on dominion/build.rs:
```diff
@@ -0,0 +1,10 @@
+use scraper::*;
+
+fn main() {
```
When does this run?

mickvangelderen replied:
`build.rs` is the default build file path. I should narrow down the `rerun-if-changed` variable though.

See https://doc.rust-lang.org/cargo/reference/build-scripts.html

Specifically:
> rerun-if-changed=PATH is a path to a file or directory which indicates that the build script should be re-run if it changes (detected by a more-recent last-modified timestamp on the file). Normally build scripts are re-run if any file inside the crate root changes, but this can be used to scope changes to just a small set of files. (If this path points to a directory the entire directory will not be traversed for changes -- only changes to the timestamp of the directory itself (which corresponds to some types of changes within the directory, depending on platform) will trigger a rebuild. To request a re-run on any changes within an entire directory, print a line for the directory and another line for everything inside it, recursively.) Note that if the build script itself (or one of its dependencies) changes, then it's rebuilt and rerun unconditionally, so cargo:rerun-if-changed=build.rs is almost always redundant (unless you want to ignore changes in all other files except for build.rs).

oliverlee commented on scraper/src/lib.rs:
```diff
@@ -0,0 +1,30 @@
+#[derive(Serialize, Deserialize, Debug)]
+pub enum Cost {
+    None,
+    Copper(u8),
```
Should this be Copper? I think the term used in manual is 'worth' which I'm not a big fan of.

mickvangelderen replied:
For now I believe `Copper` is the smallest existing unit of money and it makes sense to express costs in terms of it. I need to learn about the other expansions before I can make a good suggestion, for now Copper seems reasonable.

oliverlee commented on scraper/src/bin/scrape.rs:
```diff
@@ -0,0 +1,100 @@
+use regex::Regex;
```
👍

oliverlee reviewed (changes requested):
Looks great. Please fix the build errors.

Also add yourself to CODEOWNERS if you want to be a reviewer by default.

mickvangelderen commented:
> Looks good, but I don't feel qualified to review this :/

You are automatically marked as a reviewer. @andebjor feel free to ask questions if you like. In general the code is not as clean as it could be but I don't want to spend too much time on it because its only used for data collection very occasionally and likely to change every time we do so.

mickvangelderen committed:
Generate CardKind

Implements name() and is_action(). No more time right now to do the other
methods.

oliverlee committed:
Update crate authors and repo code owners

oliverlee committed:
Set default crate for cargo run

oliverlee committed:
Generate cards only for the base set

mickvangelderen committed:
Add Copy and Clone where reasonable

mickvangelderen committed:
Simplify build script
